sktime/forecasting/nnetsaucemts.py

```diff
@@ -0,0 +1,738 @@
+            cluster_encode=self.cluster_encode,
+            type_clust=self.type_clust,
+            type_scaling=self.type_scaling,
+            lags=self.lags,
+            type_pi=self.type_pi,
+            block_size=self.block_size,
+            replications=self.replications,
+            kernel=self.kernel,
+            agg=self.agg,
+            seed=self.seed,
+            backend=self.backend,
+            verbose=self.verbose,
+            show_progress=self.show_progress,
+        )
+        # IMPORTANT: the self.params should never be overwritten or mutated from now on
+        # for handling defaults etc, write to other attributes, e.g., self._parama
+
+        # todo: optional, parameter checking logic (if applicable) should happen here
+        # if writes derived values to self, should *not* overwrite self.parama etc
+        # instead, write to self._parama, self._newparam (starting with _)
+
+    # todo: implement this, mandatory
+    def _fit(self, y, X=None, fh=None):
+        """Fit forecaster to training data.
+
+        private _fit containing the core logic, called from fit
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        y : pd.DataFrame
+            if self.get_tag("scitype:y")=="univariate":
+                guaranteed to have a single column
+            if self.get_tag("scitype:y")=="both": no restrictions apply
+        X : pd.DataFrame, optional (default=None)
+            Exogeneous time series to fit to.
+
+        Returns
+        -------
+        self : reference to self
+        """
+        self.fitter.fit(y)
+        self.obj = self.fitter.obj
+        return self
+
+        # IMPORTANT: avoid side effects to y, X, fh
+        #
+        # Note: when interfacing a model that has fit, with parameters
+        #   that are not data (y, X) or forecasting-horizon-like,
+        #   but model parameters, *don't* add as arguments to fit, but treat as follows:
+        #   1. pass to constructor,  2. write to self in constructor,
+        #   3. read from self in _fit,  4. pass to interfaced_model.fit in _fit
+
+    # todo: implement this, mandatory
+    def _predict(self, fh, X=None):
+        """Forecast time series at future horizon.
+
+        private _predict containing the core logic, called from predict
+
+        State required:
+            Requires state to be "fitted".
+
+        Accesses in self:
+            Fitted model attributes ending in "_"
+            self.cutoff
+
+        Parameters
+        ----------
+        fh : guaranteed to be ForecastingHorizon or None, optional (default=None)
+            The forecasting horizon with the steps ahead to to predict.
+        X : pd.DataFrame, optional (default=None)
+            Exogenous time series
+
+        Returns
+        -------
+        y_pred : pd.DataFrame
+            Point predictions
+        """
+        h = fh[-1]
```

Comment:
I am not sure if this is safe. `fh` need not be sorted, for example one can pass `fh=[2,3,1]`.